src/main/java/com/truelayer/java/paymentsproviders/entities/SearchPaymentProvidersRequest.java

```diff
@@ -0,0 +1,29 @@
+package com.truelayer.java.paymentsproviders.entities;
+
+import com.truelayer.java.entities.CurrencyCode;
+import com.truelayer.java.payments.entities.CountryCode;
+import com.truelayer.java.payments.entities.CustomerSegment;
+import com.truelayer.java.payments.entities.ReleaseChannel;
+import java.util.List;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+
+@Builder
+@Getter
+@ToString
+@EqualsAndHashCode
+public class SearchPaymentProvidersRequest {
+    private List<CountryCode> countries;
+
+    private List<CurrencyCode> currencies;
+
+    private ReleaseChannel releaseChannel;
+
+    private List<CustomerSegment> customerSegments;
+
+    private Capabilities capabilities;
```

Comment:
I decided to reuse this and nested objects even if in the request `bank_transfer`, `vrp_sweeping` and `vrp_commercial` are empty, whereas in the response they contain data. 
I tried to make the request setting those extra fields and the behaviour on PGW side didn't change, so I think it's safe to assume they're ignored even if a client decides to set them. 

Anyway, if you think this is confusing I can create different objects for the request